src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -164,20 +171,87 @@ struct
                          applying the diff does not match blockchain state's \
                          ledger hash and staged ledger hash resp.\n")))
           in
-          {transition_with_hash; staged_ledger= transitioned_staged_ledger} )
+          { transition_with_hash
+          ; staged_ledger= transitioned_staged_ledger
+          ; ledger_proof_txns } )
 
     let hash {transition_with_hash; _} = With_hash.hash transition_with_hash
 
     let parent_hash {transition_with_hash; _} =
       Consensus.Protocol_state.previous_state_hash
-        (Inputs.External_transition.Verified.protocol_state
-           (With_hash.data transition_with_hash))
+        ( With_hash.data transition_with_hash
+        |> Inputs.External_transition.Verified.protocol_state )
+
+    (** Given:
+     *
+     *       bads
+     *        o       o
+     *       /       /
+     *    o ---- o -
+     *  root \  heir \
+     *        o       o
+     *               heir_heirs
+     *
+     *  commit heir into root and reparent heir_heirs onto root throw away bads
+    *)
+    let commit ~root ~heir ~bads ~heir_heirs :
```

Comment:
I'm not so sure you should separate all of these parameters. `bads`, and `heir_heirs` are computable given the `heir`. Root is also always known. Separating these parameters requires that functions which call this routine must behave correctly or they will cause bugs. I think it would be better to encapsulate more of connected state here as long as you can compute it efficiently.